### PR TITLE
Remove print and allow for grad directly on the predict method

### PR DESCRIPTION
**Pull request by borisbolliet**
Remove long print here: 
```
Tried to load pickle file from pre-trained model, but failed
...
```

(maybe make them optional?)

Exception handling to take gradients directly on predict method:

```
            try:
                return np.stack([input_dict[k] for k in self.parameters], axis=1)
            except TracerArrayConversionError:
                converted_dict = {k: jnp.array(v) if isinstance(v, list) else v for k, v in input_dict.items()}
                return jnp.stack([converted_dict[k] for k in self.parameters], axis=1)
```

### CODE REVIEW

**Comment by dpiras** [comment]
Thank you @borisbolliet!

Regarding the first, I'm a but on the fence. Maybe we should add a `verbose` flag that is on by default, so that new users don't get discouraged if they get errors but they don't know what is going on? And advanced users can turn it off?

Regarding the second, could you please help me understand the use case? Shouldn't the `derivative` method already allow gradients of the predictions?

**Comment by borisbolliet** [comment]
Right, yes I think a verbose flag would be great!
Regarding second part, is for when we want to use the emulators loaded via cosmopower_jax, but in codes outside cosmopower_jax, when emulators are parts of other functions. Will send an explicit example. (But maybe you are right that all can be done via derivative method and I missed it.)

**Comment by dpiras** [comment]
Hey @borisbolliet, I took the liberty to add the verbose flag directly (https://github.com/dpiras/cosmopower-jax/pull/13), feel free to use it by `pip install`ing the latest 0.5.5 version.

Let me know if the other issue is still relevant, either here or in a new PR with the explicit example you mentioned. I will close this PR by the end of this week if there is no activity, but feel free to (re-)open in case!

**Comment by borisbolliet** [comment]
Great yes let's do that! Thank you!!
